core_types/src/agent.rs

```diff
@@ -1,129 +1,64 @@
-        KeyBuffer(*b)
-    }
-
-    /// generate a key buffer from raw bytes from two parts (no correction)
-    pub fn with_raw_parts(
-        a: &[u8; KeyBuffer::HALF_KEY_LEN],
-        b: &[u8; KeyBuffer::HALF_KEY_LEN],
-    ) -> KeyBuffer {
-        let mut buf: [u8; KeyBuffer::KEY_LEN] = [0; KeyBuffer::KEY_LEN];
-
-        buf[..KeyBuffer::HALF_KEY_LEN].clone_from_slice(&a[..KeyBuffer::HALF_KEY_LEN]);
-
-        buf[KeyBuffer::HALF_KEY_LEN..KeyBuffer::KEY_LEN]
-            .clone_from_slice(&b[..KeyBuffer::HALF_KEY_LEN]);
-
-        KeyBuffer(buf)
-    }
+use std::{convert::TryFrom, str};
 
-    /// render a base64url encoded user identity with reed-solomon parity bytes
-    pub fn render(&self) -> String {
-        let enc = Encoder::new(KeyBuffer::PARITY_LEN);
-        let enc = *enc.encode(&self.0);
-        base64::encode(&enc[..]).replace("+", "-").replace("/", "_")
-    }
+use hcid::*;
 
-    /// get the signature public key portion of this buffer
-    pub fn get_sig(&self) -> &[u8; KeyBuffer::HALF_KEY_LEN] {
-        array_ref![self.0, 0, KeyBuffer::HALF_KEY_LEN]
-    }
+pub type Base32 = String;
 
-    /// get the encryption public key portion of this buffer
-    pub fn get_enc(&self) -> &[u8; KeyBuffer::HALF_KEY_LEN] {
-        array_ref![self.0, KeyBuffer::HALF_KEY_LEN, KeyBuffer::HALF_KEY_LEN]
-    }
-}
-
-/// agent data that can be stored in the cas
-/// note thate the "address" of an agent entry is the base64url encoded
-/// public key identity string
+/// AgentId represents an agent in the Holochain framework.
+/// This data struct is meant be stored in the CAS and source-chain.
+/// Its key is the public signing key, and is also used as its address.
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq, DefaultJson)]
 pub struct AgentId {
     /// a nickname for referencing this agent
     pub nick: String,
-    /// the base64url encoded public identity string for this agent
-    pub key: String,
+    /// the encoded public signing key of this agent (the magnifier)
+    pub pub_sign_key: Base32,
 }
 
 impl AgentId {
-    /// generate a fake testing agent
-    /// `s` will be used for the `nick` and included in the key string as well
-    /// this agent is not cryptographically generated...
-    /// it will not be able to sign / encrypt anything
+    /// generate a agent id with fake key
     pub fn generate_fake(nick: &str) -> Self {
-        let mut buf = nick.to_string();
-        // Make sure base64 string must is big enough to decode into 64 bytes key
-        while buf.len() < 82 {
-            buf.push_str("+");
-        }
-        buf.push_str("AAAA");
-        let buf = base64::decode(&buf)
-            .expect("could not decode the generated fake base64 string - use the base64 alphabet");
-        let buf = KeyBuffer::with_raw(array_ref![buf, 0, KeyBuffer::KEY_LEN]);
-        AgentId::new(nick, &buf)
+        let mut key: [u8; 32] = [0; 32];
+        key[0] = 42;
```

Comment:
Would it make sense to have a well-known testing keypair (i.e. the priv-key is published in the sourcecode with warnings not to really use it?)

Reply:
Probably. Changed it to a const within AgentId.